indexer.go

```diff
@@ -25,7 +32,117 @@ func NewIndexer(provider Provider, writer Writer) *Indexer {
 	}
 }
 
-// IndexBlock transfers all transactions to a known strcuture and saves them
+// Transaction struct handler of all transaction fields to be indexed
+type Transaction struct {
+	ID              int
+	Hash            string
+	FromAddress     string
+	ToAddress       string
+	AppPubKey       string
+	Blockchains     []string
+	MessageType     string
+	Height          int
+	Index           int
+	Proof           *provider.TransactionProof
+	StdTx           *provider.StdTx
+	TxResult        *provider.TxResult
+	Tx              string
+	Entropy         int
+	Fee             int
+	FeeDenomination string
+}
+
+func convertProvTransactionToTransaction(provTransaction *provider.Transaction) *Transaction {
+	var fromAddress, toAddress string
+	var blockChains []string
+
+	stdTx := provTransaction.StdTx
+	msgValues := stdTx.Msg.Value
+	feeStruct := stdTx.Fee[0]
+
+	rawFromAddress, ok := msgValues["from_address"].(string)
+	if ok {
+		fromAddress = rawFromAddress
+	}
+
+	rawToAddress, ok := msgValues["to_address"].(string)
+	if ok {
+		toAddress = rawToAddress
+	}
+
+	rawBlockChains, ok := msgValues["chains"].([]any)
+	if ok {
+		for _, rawBlockChain := range rawBlockChains {
+			blockChain, ok := rawBlockChain.(string)
+			if ok {
+				blockChains = append(blockChains, blockChain)
+			}
+		}
+	}
+
+	fee, _ := strconv.Atoi(feeStruct.Amount)
+
+	return &Transaction{
+		Hash:            provTransaction.Hash,
+		FromAddress:     fromAddress,
+		ToAddress:       toAddress,
+		AppPubKey:       stdTx.Signature.PubKey,
+		Blockchains:     blockChains,
+		MessageType:     stdTx.Msg.Type,
+		Height:          provTransaction.Height,
+		Index:           provTransaction.Index,
+		Proof:           provTransaction.Proof,
+		StdTx:           stdTx,
+		TxResult:        provTransaction.TxResult,
+		Tx:              provTransaction.Tx,
+		Entropy:         int(stdTx.Entropy),
+		Fee:             fee,
+		FeeDenomination: feeStruct.Denom,
+	}
+}
+
+// IndexBlockTransactions converts block transactions to a known structure and saves them
+func (i *Indexer) IndexBlockTransactions(blockHeight int) error {
+	// TODO: add pagination support
+	blockTransactionsOutput, err := i.provider.GetBlockTransactions(blockHeight, nil)
+	if err != nil {
+		return err
+	}
+
+	var transactions []*Transaction
+
+	for _, tx := range blockTransactionsOutput.Txs {
+		transactions = append(transactions, convertProvTransactionToTransaction(tx))
```

Comment:
Ah, so it's trying to marshal transactions. Why not `marshallTransaction` or `tryMarshallTransaction`?

Reply:
In Go, marshal is most of the time related to JSON not in between strutcts